src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/AbstractPathJob.java

```diff
@@ -499,12 +499,29 @@ private static boolean calculateSwimming(@NotNull final IWorldReader world, @Not
 
     /**
      * Check if the block at this position is actually some kind of waterly fluid.
+     *
      * @param pos the pos in the world.
      * @return true if so.
      */
     private static boolean isWater(@NotNull final IWorldReader world, final BlockPos pos)
     {
-        final BlockState state = world.getBlockState(pos);
+        return isWater(world, pos, null, null);
+    }
+
+    /**
+     * Check if the block at this position is actually some kind of waterly fluid.
+     * @param pos the pos in the world.
+     * @param state existing blockstate or null
+     * @param fluidState existing fluidstate or null
+     * @return true if so.
+     */
+    private static boolean isWater(@NotNull final IWorldReader world, final BlockPos pos, @Nullable BlockState state, @Nullable FluidState fluidState)
+    {
+        if (state == null)
```

Comment:
Uh, this is ugly, reusing a param, rather make it a temp var that gets the input from the param =D